lib/parse.js

```diff
@@ -436,6 +438,35 @@ function tokenizer($TEXT, filename, html5_comments, shebang) {
         return tok;
     });
 
+    var read_template_characters = with_eof_error("SyntaxError: Unterminated template", function(begin){
+        if (begin) {
+            S.template_braces.push(S.brace_counter);
+        }
+        var content = "", ch, tok;
+        next();
+        while ((ch = next(true)) !== "`") {
+            if (ch === "$" && peek() === "{") {
+                next();
+                S.brace_counter++;
+                tok = token(begin ? "template_head" : "template_substitution", content);
+                tok.begin = begin;
+                tok.end = false;
+                return tok;
+            }
+
+            if (ch === "\\") {
+                ch = read_escaped_char();
```

Comment:
Once #1164 gets merged to harmony, this should be able to handle octal escaping as well.